src/games/tritium.ts

```diff
@@ -317,23 +323,31 @@ export class TritiumGame extends GameBase {
         return result;
     }
 
-    public move(m: string, {trusted = false} = {}): TritiumGame {
+    public move(m: string, {partial = false, trusted = false} = {}): TritiumGame {
         if (this.gameover) {
             throw new UserFacingError("MOVES_GAMEOVER", i18next.t("apgames:MOVES_GAMEOVER"));
         }
 
-        if (! trusted) {
+        if (!trusted) {
             const result = this.validateMove(m);
-            if (! result.valid) {
+            if (!result.valid) {
                 throw new UserFacingError("VALIDATION_GENERAL", result.message)
             }
-            if (! this.moves().includes(m)) {
+            if (!partial && !this.moves().includes(m)) {
                 throw new UserFacingError("VALIDATION_FAILSAFE", i18next.t("apgames:validation._general.FAILSAFE", {move: m}))
             }
         }
 
         this.results = [];
 
+        if (m.length === 0) { return this; }
+        if (partial) {
+            this.selected = m;
+            this.results.push({ type: "select", what: this.selected });
+            return this;
+        }
+        this.selected = undefined;
+
```

Comment:
Something that doesn't appear in your commit is the early `return` necessary when processing partial moves. *After* processing the move, but *before* updating currplayer, checking for EOG, and adding the state to the stack, you need the following:

```ts
if (partial) { return this; }
```

You will see examples of this in other game code. The reason for this is that `checkEOG` modifies the object beyond just the stack entry. You don't want that happening until the move is final. This is the result of some admittedly poor design decisions made early on, but changing them at this point is nontrivial.

Reply:
Mmm, not sure I get that. The early return is handled right there line 347, in case of a partial move. There's nothing to process in that case: no state change, only the selection for the UI's sake.
Only when the move is complete the rest of the function happens, with the state change and production the `results`.

Reply:
Oh! I might have just missed it. Apologies. As long as you're returning before those last few function calls, we're good.